### ISSUE TITLE
HPWH additional fields

### ISSUE THREAD

**Issue by shorowit** [hpxmlwg/hpxml#404]
Some way to describe:
- 120V products (could apply to other appliances like dryers, cooking range, etc. too)
- Shared vs dedicated circuit
- Tempering valve (integrated vs separate, temperature, etc.)
- ???

**Comment by jmaguire1** [comment]
There's been a few offerings of low power appliances that can run on 120V for the residential market, as a way to electrify without requiring a panel upgrade. Currently, there's three products types of products I'm aware of that have 120V offerings (or will soon):

- [Clothes dryers](https://www.homedepot.com/b/Appliances-Washers-Dryers-Dryers-Electric-Dryers/120-volt/N-5yc1vZc3q1Z1z17ja4)
- Cooking [ranges](https://copperhome.com/)
- Heat pump water [heaters](https://www.rheem.com/water-heating/articles/rheem-proterra-plug-in-heat-pump-water-heaters-designed-for-easy-gas-unit-replacement-2/)

I'm currently tackling 120V HPWH modeling for OS-HPXML, and trying to think of the best way to designate this in HPXML. I think the safest thing would be an explicit 120V/240V element associated with the HPWH object. You could try to infer this from things like the element/HP capacity, but that seems a lot less straightforward than directly identifying the power draw.

To make things a little more complicated, there's also "shared" and "dedicated" circuit versions of 120V HPWHs on the market, at least from one manufacturer. My understanding is per NEC to be "shared" max power draw has to be < 900 W (half of a 15A 120V circuit), which limits how quickly you can reheat the tank. Lab testing shows this matters a lot in terms of recovery time, so it probably makes sense to install a dedicated circuit version if you have the capacity (which is probably the minority of homes).

![image](https://github.com/user-attachments/assets/83d9fe28-06b1-41cb-9d02-360152715d8e)

To deal with that, maybe we have an optional "shared"/"dedicated" option if it is 120V, with the default being shared if not provided? Shared is more what people think of when they think of 120V appliances, but the extra recovery could be a big deal and so if I were looking to do this and had the panel capacity, I'd probably opt for a dedicated circuit version.

The one other thing we'll want to deal with as part of modeling 120V HPWHs is that they frequently overheat (to 140F) and include a mixing valve with a separate setpoint (125F by default). I see HPXML has a "[HasMixingValve](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HasMixingValve)" element along with "[HotWaterTemperature](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HotWaterTemperature)", but is there any way to set the mixing valve temperature? We could default it to 125F, but my understanding is users can actually change the mixing valve temperature, which is closer to what they get out of the tap, rather than the setpoint temperature to boost the energy stored and make a cold shower less likely. Is it worth adding a mixing valve setpoint if a mixing valve is specified, rather than just saying one exists and assuming it's setpoint?

**Comment by lixiliu** [comment]
Shared / dedicated circuit designation can be used to inform breaker space requirement, and branch circuit amp if we need this in the future.

**Comment by jmaguire1** [comment]
Right, so we'll want that designation. Not every manufacturer has shared/dedicated (some only offer shared), so we should make it optional and/or default to shared if not provided.

**Comment by jmaguire1** [comment]
Proposed changes:

New Element: **HPWH Voltage**
Similar to [HPWH Operation Mode](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HPWHOperatingMode) in that it's only applicable to HPWHs. Options are 240V/120V, default is 240V.

New Element: **120V HPWH Shared/Dedicated Circuit**
Similar to [HPWH Operation Mode](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HPWHOperatingMode) in that it's only applicable to HPWHs. Options are "Dedicated"/"Shared", default is "Shared". Ideally, this one would be dependent on being a 120V HPWH, since "Shared" and "240V" aren't compatible options, only 120V HPWHs have this distinction. 

@nmerket: Is it possible to propose something that's a sub element of a new element like that, so you could never define a 240V Shared option? I made the name "120V..." to try to make it clearer as well.

**Comment by shorowit** [comment]
@jmaguire1 What if the two elements were combined so that the options are, e.g.:
- `240V`
- `120V` (shared/dedicated circuit unknown)
- `120V, dedicated circuit`
- `120V, shared circuit`

Then you wouldn't have the problem you described.

**Comment by jmaguire1** [comment]
@shorowit: Sounds good to me, especially if that makes the implementation easier! I'll leave the old one in for posterity, but revised proposal:

New Element: **HPWH Voltage**
Similar to [HPWH Operation Mode](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HPWHOperatingMode) in that it's only applicable to HPWHs. Options are:

- 240V
- 120V (unknown shared/dedicated)
- 120V, dedicated circuit
- 120V, shared circuit

For reference I think if unknown and anyone has to guess, "shared" is the better guess, that's what people are generally thinking of when they picture a 120V HPWH. Dedicated circuit products do exist, but you need an unused 15A circuit to be able to take advantage of them.

**Comment by nmerket** [comment]
That makes sense to me.

**Comment by jmaguire1** [comment]
One more proposed change: Right now HPXML has a boolean field for [has mixing valve](https://hpxml.nrel.gov/datadictionary/4.0.0/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem/HasMixingValve). I think we ought to also have a temperature setpoint for the mixing valve. For a 120V HPWH, I think we'd generally be pretty safe assuming if you have a mixing valve, it's set to 125 F, but someone could theoretically change this. Mixing valves are also required for WH DR programs to prevent burns during "advanced load add" events, so it might not be as obvious in that scenario what the mixing valve is set to. Especially if you're only doing add OR shed events, not both.

@nmerket: Since the existing variable is just a bool, what's the easiest approach here? Change it to "mixing valve setpoint", and if it's not there then there's no mixing valve?

**Comment by nmerket** [comment]
The best thing would be to add a new `MixingValveSetpoint` in addition to the bool element. That way we're not breaking backwards compatibility.
